segment.go

```diff
@@ -149,3 +153,26 @@ func (segment *Segment) mapSegmentWithWriter(env environmentInfo) error {
 	}
 	return errors.New("unable to map writer")
 }
+
+func (segment *Segment) setStringValue(env environmentInfo, cwd string, debug bool) {
+	err := segment.mapSegmentWithWriter(env)
+	if err != nil || segment.hasValue(IgnoreFolders, cwd) {
+		return
+	}
+	// add timing only in debug
+	if debug {
+		start := time.Now()
+		defer (func() {
+			// force segment rendering to display the time it took
+			// to check if the segment is enabled or not
+			// depending on the segement, calling enabled()
+			// can be time consuming
+			segment.active = true
```

Comment:
Can you elaborate why you'd want to do this? As it won't necessarily run the logic (which is what you want to time).

Reply:
because for some segments, calling `enabled()` is time consuming even if it returns false(ie. git segment).
This way we have all configured segments from the json and their timings. Otherwise we can miss some informations.